RFunction.R

```diff
@@ -7,279 +7,259 @@ library(magic)
 library(geosphere)
 library(lubridate)
 
-rFunction <-function(data, threshold=NULL, window=72, yaxs_limit=1000){
-  
-   data <- data |> mutate(location_long = sf::st_coordinates(data)[,1],
-                         location_lat = sf::st_coordinates(data)[,2],
-                         trackID = mt_track_id(data),
-                         distance = mt_distance(data))
-
-   #data$distance <- mt_distance(data, units="m") # like this units will always be in meters, please adjust to the units you see most fit
-  class(data$distance) <-"numeric"
-  data_df <-as.data.frame(data)
-  names(data_df) <- gsub("[.]", "_", names(data_df))
-  # uid <-unique(data_df$individual_local_identifier)
-  uid <-unique(data$trackID) 
-  
-  ###  Function for plotting the individual speed
-  plot_speed <-function(dat, dat_outp, yul=yaxs_limit)
-  {
-    yr <- year(dat$timestamp[1])
-    plot(dat$timestamp, dat$speed, main= paste(uid[i], yr, sep = "_"), cex=0.4, ylim=c(0,yul),
-         ylab= expression(paste("Distance /", Delta, "t")), xlab= "Time", col= "grey40")
-    lines(dat$timestamp, dat$speed,col= "grey30", main= paste(uid[i], yr, sep = "_"))
-    lines(dat$timestamp, dat$rollm, col ="brown4", lwd=1.5, main= paste(uid[i], yr, sep = "_"))
-    #legend('topright', legend = rp, bty = 'n')
-    abline(h= ifelse(is.null(threshold), mean(dat$speed, na.rm=T), threshold),  lty=3, lwd=2, col= "coral")
-    for (i in 1:nrow(dat_outp))
-    {abline(v= dat_outp$V5, lty=2, lwd=1.5, col= "green4")
-      abline(v= dat_outp$V6, lty=4, lwd=1.5, col= "royalblue")
-    }
+
```

Comment:
Move plotting functions outside of main function to make main function more readable